haskell/private/ls_modules.py

```diff
@@ -57,10 +54,36 @@
     # Substitute package ids for package names in reexports, because
     # GHC really wants package ids.
     regexp = re.compile("from (%s):" % "|".join(map(re.escape, pkg_ids_map)))
-    reexported_modules = (
-        regexp.sub(lambda match: "from %s:" % pkg_ids_map[match.group(1)], mod)
+
+    # Any duplicate used during the replacement phase will trigger a
+    # script failure later
+    duplicate_used = set()
+
+    def replace_pkg_by_pkgid(match):
+        pkgname = match.group(1)
+
+        if pkgname in duplicates:
+            duplicate_used.add(pkgname)
+            return "" # arbitrary return value, the script will crash later
```

Comment:
Why not fail right then and here?

Reply:
Bette user experience. If there are many *used* duplicates, it will fail and report all used duplicates, instead of the first one.

That's a tradeoff, but I can simplify the source code and fail on the first error.

Reply:
I think it's better to simplify the logic. It is unlikely that there will be many duplicate modules to deal with, so it's not onerous for the user to print only the first one. I think you can even get rid of the inner function definition this way.

Reply:
@mboes I fixed the logic. However I don't understand when you say:

> I think you can even get rid of the inner function definition this way

If you are talking about the newly introduced `replace_pkg_by_pkgid`, you are right that everything can be put inside a `lambda`, but from my point of view this will lead to an unreadable chunk of code. In case you were meaning something else, could you detail?